app/models/miq_server/worker_management/systemd.rb

```diff
@@ -1,7 +1,7 @@
 class MiqServer::WorkerManagement::Systemd < MiqServer::WorkerManagement
   def sync_from_system
-    self.miq_services = systemd_services.select { |unit| manageiq_service?(unit) }
-    self.miq_services_by_guid = miq_services.index_by { |w| w[:name].match(/^.+@(?<guid>.+)\.service$/).named_captures["guid"] }
```

Comment:
NOTE this line in particular meant that we _had_ to have workers with `@guid.service` even if there could only be a single instance.  By having the unit name available in the worker record it frees us up to name worker services by other patterns.